src/Event/EventSubscriber/SignatureSubscriber.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Event\EventSubscriber;
+
+use App\Security\Request\RequestVerifierInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+use Symfony\Component\HttpKernel\Exception\BadRequestHttpException;
+use Symfony\Component\HttpKernel\KernelEvents;
+use Symfony\Component\Security\Core\Security;
+
+class SignatureSubscriber implements EventSubscriberInterface
+{
+    /**
+     * @var Security
+     */
+    private $security;
+
+    /**
+     * @var RequestVerifierInterface
+     */
+    private $requestVerifier;
+
+    public function __construct(Security $security, RequestVerifierInterface $requestVerifier)
+    {
+        $this->security = $security;
+        $this->requestVerifier = $requestVerifier;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function getSubscribedEvents()
+    {
+        return [
+            KernelEvents::REQUEST => 'onKernelRequest',
+        ];
+    }
+
+    public function onKernelRequest(GetResponseEvent $event)
+    {
+        $request = $event->getRequest();
+        if (('false' === getenv('AVAILABLE_REQUEST_SIGNATURE') && 'false' === getenv('STRICT_AUDIT_TRAIL'))
```

Comment:
Maybe send flags to constructor instead hardcode env variables in code?

Reply:
What for?

Reply:
To avoid hardcode environment variables in the code.

Reply:
Up to you..